org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/MoveHandlerTest.java

```diff
@@ -163,7 +163,7 @@ public void testMoveFile() throws JavaModelException, BadLocationException {
 		assertNotNull(refactorEdit);
 		assertNotNull(refactorEdit.edit);
 		List<Either<TextDocumentEdit, ResourceOperation>> changes = refactorEdit.edit.getDocumentChanges();
-		assertEquals(4, changes.size());
+		assertEquals(3, changes.size());

```

Comment:
why would this change?

Reply:
The move refactoring code generates some empty textEdit operation, for example, insert empty string at the same location. I refined the ChangeUtils utility to filter out such empty operations. That's why you see the changes list is shorter.